src/net_processing.cpp

```diff
@@ -1451,12 +1451,12 @@ bool PeerManagerImpl::MaybePunishNodeForTx(NodeId nodeid, const TxValidationStat
     return false;
 }
 
-bool PeerManagerImpl::BlockRequestAllowed(const CBlockIndex* pindex)
+bool PeerManagerImpl::BlockRequestAllowed(const CBlockIndex* pindex, bool check_valid_scripts)
 {
     AssertLockHeld(cs_main);
     if (m_chainman.ActiveChain().Contains(pindex)) return true;
-    return pindex->IsValid(BLOCK_VALID_SCRIPTS) && (pindexBestHeader != nullptr) &&
-           (pindexBestHeader->GetBlockTime() - pindex->GetBlockTime() < STALE_RELAY_AGE_LIMIT) &&
+    return (!check_valid_scripts || pindex->IsValid(BLOCK_VALID_SCRIPTS)) &&
```

Comment:
Seems like a good refactor candidate, as these conditions are hard to follow and review.

Reply:
Good point, i added a refactor commit that also adds some comments.